gallery_dl/extractor/common.py

```diff
@@ -164,7 +164,8 @@ def request(self, url, *, method="GET", session=None,
                     if b'name="captcha-bypass"' in content:
                         self.log.warning("Cloudflare CAPTCHA")
                         break
-                if code not in retry_codes and code < 500:
+                if code not in retry_codes and code < 500 and not \
+                        (retry_on and retry_on(response)):
                     break
```

Comment:
```suggestion
                if code not in retry_codes and code < 500:
                    break
                if retry_on and not retry_on(response):
                    break
```

Just use two `if`s.
Looks better and makes it easier to appease flake8.

Reply:
`retry_on` will not be reachable unless `403` is in `retry_codes`. Perhaps you meant this:
```python
                if code not in retry_codes and code < 500:
                    if not (retry_on and retry_on(response)):
                        break
```